### DISCUSSION TITLE
Order by datetime (only date part)

### DISCUSSION THREAD

**Discussion by homawog** [ujjwalguptaofficial/JsStore#254] [Q&A] [answered]
I want to sort a table by two fields, first by a column in DateTime format, then by a column in Number format. This is actually no problem, but I am only interested in the date part of the column, i.e. if there are several entries per day, the second column should be the first criterion.
In SQL I get the desired result with AS, CONVERT and CAST, but in JSStore I haven't found a way yet.
Maybe someone has a tip for me?

Thanks a lot
Martin

**Comment by ujjwalguptaofficial** [comment]
It's hard to understand what you are saying, may be if you can provide a something reproducable then i can help you.

**Comment by homawog** [comment]
Hello Ujjwal,

thank you for your reply and also for providing JsStore, which makes using IndexedDB so much easier.
I have attached here a short description of my problem and my solution in MySQL.
[Description.pdf](https://github.com/ujjwalguptaofficial/JsStore/files/7788365/Description.pdf)

**Comment by homawog** [comment]
Hello Ujjwal,
thank you for your reply and also for providing JsStore, which makes using IndexedDB so much easier.I have attached here a short description of my problem and my solution in MySQL.
I have entered this answer and the description also in the Github discussion

Best regardsMartin Hofer


   Am Mittwoch, 29. Dezember 2021, 03:25:27 MEZ hat Ujjwal Gupta ***@***.***> Folgendes geschrieben:  
 
 


It's hard to understand what you are saying, may be if you can provide a something reproducable then i can help you.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ujjwalguptaofficial** [comment]
Could you share the data with me in json file, so that i can try. or better if you can create the repo and give me access.

**Comment by homawog** [comment]
Hi Ujjwal,
I share the data with the json file atr.json.
I am unfortunately relatively new to Github and have never created a repo before. I will take a closer look to understand what exactly is being done here and then give it a try.

Thanks for your help and kind regards
Martin

    Am Freitag, 31. Dezember 2021, 12:42:28 MEZ hat Ujjwal Gupta ***@***.***> Folgendes geschrieben:  
 
 


Could you share the data with me in json file, so that i can try. or better if you can create the repo and give me access.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by homawog** [comment, marked as answer]
Hello again Ujjwal,
I have now found a solution to my problem. This is not particularly clean programmatically, since a data redundancy arises, but I can live with it ;-)
I have added an additional data field with the type number and store there the respective date as number in the form YYYYMMDD. This way I can sort the data as desired, i.e. first today's date, then priority.
Thank you very much for your interest, happy new year and thanks again for your cool plugin, which makes working with IndexDB so much easier.
Martin
